src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -509,9 +510,12 @@ private async Task<TextDocument> UpdateDocumentInfoAsync(TextDocument document,
             private async Task ValidateChecksumAsync(
                 Checksum checksumFromRequest,
                 Solution incrementalSolutionBuilt,
+                ProjectId? projectConeId,
                 CancellationToken cancellationToken)
             {
-                var currentSolutionChecksum = await incrementalSolutionBuilt.CompilationState.GetChecksumAsync(CancellationToken.None).ConfigureAwait(false);
+                var currentSolutionChecksum = projectConeId == null
+                    ? await incrementalSolutionBuilt.CompilationState.GetChecksumAsync(CancellationToken.None).ConfigureAwait(false)
+                    : await incrementalSolutionBuilt.CompilationState.GetChecksumAsync(projectConeId, CancellationToken.None).ConfigureAwait(false);
```

Comment:
in the case of a cone sync, we only want to compare the checksum of the cone sync'ed over to the current checksum of that same cone.  what's outside of those cones is totally allowed to be different.